posthog/api/github.py

```diff
@@ -0,0 +1,162 @@
+    except InvalidSignature as e:
+        raise SignatureVerificationError("Signature does not match payload") from e
+
+
+class SecretAlertSerializer(serializers.Serializer):
+    token = serializers.CharField()
+    type = serializers.ChoiceField(choices=["posthog_personal_api_key", "posthog_feature_flags_secure_api_key"])
+    url = serializers.CharField(allow_blank=True)
+    source: Any = serializers.CharField()
+
+
+class SecretAlert(APIView):
+    authentication_classes = []
+    permission_classes = [AllowAny]
+    parser_classes = [JSONParser]
+
+    def post(self, request):
+        kid = (request.headers.get("Github-Public-Key-Identifier") or "").strip()
+        sig = (request.headers.get("Github-Public-Key-Signature") or "").strip()
+
+        if not kid:
+            raise ValidationError(
+                {
+                    "headers": {
+                        "Github-Public-Key-Identifier": "required non-blank string",
+                    }
+                }
+            )
+        if not sig:
+            raise ValidationError(
+                {
+                    "headers": {
+                        "Github-Public-Key-Signature": "required non-blank string",
+                    }
+                }
+            )
+
+        try:
+            verify_github_signature(request.body.decode("utf-8"), kid, sig)
+        except SignatureVerificationError:
+            return Response({"detail": "Invalid signature"}, status=401)
+
+        if not isinstance(request.data, list):
+            raise ValidationError(detail="Expected a JSON array")
+        if len(request.data) < 1:
+            raise ValidationError(detail="Array must contain at least one item")
+
+        secret_alert = SecretAlertSerializer(data=request.data, many=True)
+        secret_alert.is_valid(raise_exception=True)
+        items = secret_alert.validated_data
+
+        results = []
+        for _, item in enumerate(items):
+            result = {
+                "token_hash": sha256(item["token"].encode("utf-8")).hexdigest(),
+                "token_type": item["type"],
+                "label": "false_positive",
+            }
+
+            if item["type"] == "posthog_personal_api_key":
+                key_lookup = find_personal_api_key(item["token"])
+                if key_lookup is not None:
+                    result["label"] = "true_positive"
+                    more_info = f"This key was detected by GitHub at {item['url']}."
+
+                    # roll key
+                    key, _ = key_lookup
+                    old_mask_value = key.mask_value
+                    serializer = PersonalAPIKeySerializer(instance=key)
+                    serializer.roll(key)
+                    send_personal_api_key_exposed(key.user.id, key.id, old_mask_value, more_info)
+
+            elif item["type"] == "posthog_feature_flags_secure_api_key":
+                Team = apps.get_model(app_label="posthog", model_name="Team")
```

Comment:
Any reason you're not doing this the easy way around?

```suggestion
                from posthog.models import Team
```

Reply:
I was copying the below code. Are those lines functionally equivalent?

 https://github.com/PostHog/posthog/blob/b8d7aaabd64cee1cf81674d954e0d9fc636244cb/posthog/auth.py#L231

Reply:
Tested and they are indeed functionally equivalent. Switched to the more direct import.